From 4745ef91c78d4d4eee4b52d78bd5d6e83445e7ce Mon Sep 17 00:00:00 2001
From: Tobias Loew <tobias.loew@steag.com>
Date: Mon, 29 Oct 2018 09:25:16 +0100
Subject: [PATCH 1/3] removed deprecated std::allocator<void>

replaced std::allocator<void> by std::allocator<char>
use std::allocator_traits for rebinding
---
 include/boost/bimap/detail/bimap_core.hpp                 | 6 +++++-
 .../boost/bimap/detail/manage_additional_parameters.hpp   | 8 ++++----
 2 files changed, 9 insertions(+), 5 deletions(-)

diff --git a/boost/bimap/detail/bimap_core.hpp b/boost/bimap/detail/bimap_core.hpp
index f642e96..c821a30 100644
--- a/boost/bimap/detail/bimap_core.hpp
+++ b/boost/bimap/detail/bimap_core.hpp
@@ -404,9 +404,13 @@ class bimap_core
     <
         relation,
         core_indices,
+#ifdef BOOST_NO_CXX11_ALLOCATOR
         BOOST_DEDUCED_TYPENAME parameters::allocator::
             BOOST_NESTED_TEMPLATE rebind<relation>::other
-
+#else
+		BOOST_DEDUCED_TYPENAME std::allocator_traits< BOOST_DEDUCED_TYPENAME parameters::allocator >::
+			BOOST_NESTED_TEMPLATE rebind_alloc<relation>
+#endif
     > core_type;
 
     // Core metadata
diff --git a/boost/bimap/detail/manage_additional_parameters.hpp b/boost/bimap/detail/manage_additional_parameters.hpp
index d95a4f0..78d89ae 100644
--- a/boost/bimap/detail/manage_additional_parameters.hpp
+++ b/boost/bimap/detail/manage_additional_parameters.hpp
@@ -83,7 +83,7 @@ struct manage_additional_parameters
     struct case_NNN
     {
         typedef left_based set_type_of_relation;
-        typedef std::allocator<void> allocator;
+        typedef std::allocator<char> allocator;
         typedef ::boost::mpl::na additional_info;
     };
 
@@ -110,7 +110,7 @@ struct manage_additional_parameters
     struct case_SNN
     {
         typedef AP1 set_type_of_relation;
-        typedef std::allocator<void> allocator;
+        typedef std::allocator<char> allocator;
         typedef ::boost::mpl::na additional_info;
     };
 
@@ -137,7 +137,7 @@ struct manage_additional_parameters
     struct case_HNN
     {
         typedef left_based set_type_of_relation;
-        typedef std::allocator<void> allocator;
+        typedef std::allocator<char> allocator;
         typedef BOOST_DEDUCED_TYPENAME AP1::value_type additional_info;
     };
 
@@ -151,7 +151,7 @@ struct manage_additional_parameters
     struct case_SHN
     {
         typedef AP1 set_type_of_relation;
-        typedef std::allocator<void> allocator;
+        typedef std::allocator<char> allocator;
         typedef BOOST_DEDUCED_TYPENAME AP2::value_type additional_info;
     };
 

From 747a9df3d41fb0fd9f53793c0c5552cd24c38717 Mon Sep 17 00:00:00 2001
From: Tobias Loew <tobias.loew@steag.com>
Date: Thu, 8 Nov 2018 17:14:56 +0100
Subject: [PATCH 2/3] Update include/boost/bimap/detail/bimap_core.hpp

---
 include/boost/bimap/detail/bimap_core.hpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/boost/bimap/detail/bimap_core.hpp b/boost/bimap/detail/bimap_core.hpp
index c821a30..e219f4e 100644
--- a/boost/bimap/detail/bimap_core.hpp
+++ b/boost/bimap/detail/bimap_core.hpp
@@ -18,6 +18,7 @@
 
 #include <boost/config.hpp>
 
+#include <memory>
 #include <boost/mpl/placeholders.hpp>
 #include <boost/mpl/push_front.hpp>
 #include <boost/mpl/if.hpp>

From 391e1d422199ab7a529789f10483ed75a7fe9808 Mon Sep 17 00:00:00 2001
From: Tobias Loew <tobias.loew@steag.com>
Date: Thu, 8 Nov 2018 17:17:57 +0100
Subject: [PATCH 3/3] Update include/boost/bimap/detail/bimap_core.hpp

---
 include/boost/bimap/detail/bimap_core.hpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/boost/bimap/detail/bimap_core.hpp b/boost/bimap/detail/bimap_core.hpp
index e219f4e..be0e1ba 100644
--- a/boost/bimap/detail/bimap_core.hpp
+++ b/boost/bimap/detail/bimap_core.hpp
@@ -18,7 +18,9 @@
 
 #include <boost/config.hpp>
 
+#ifndef BOOST_NO_CXX11_ALLOCATOR
 #include <memory>
+#endif
 #include <boost/mpl/placeholders.hpp>
 #include <boost/mpl/push_front.hpp>
 #include <boost/mpl/if.hpp>
